install postgresql 9.6 from the official postgres repo on redhat7 and…

lhoss opened the pull request:
… redhat-amazon (to fix SchemaRegistry on HDP-3)

fix for https://github.com/hortonworks/ansible-hortonworks/issues/61

Notes: 
* I made it generic enough (using postgres_version variable similar to existing mysql_version), so it's also much easier to update for ex to postgres 10.2.x
* The current PR handles both redhat/centos-7 and redhat-Amazon (and using the exact same config variables)
  * tests were done (only ensured Ambari works) on HDP-2.6.5 (as HDP-3.0.x full stack test with postgres v9.6, done already)
  * probably these generic variables should also work for  redhat/centos-6 (but not tested) 
* following cmd, I had used in my first centos7 try, did not work anymore on Amazon linux, so I replaced by using directly the more standard 'initdb' cmd (used by most postgres roles):
>  postgres_initdb_command: postgresql96-setup initdb
* The test I did on Amazon (with working postgres 9.6 install), was using a recent Amazon-linux-2
(us-west-2    | amzn2-ami-hvm-2.0.20180810-x86_64-ebs | ami-99dafbe1), but starting up Ambari (HDP 2.6.5) failed with:
>  File "/usr/lib/ambari-agent/lib/ambari_commons/os_check.py", line 311, in _get_os_version   raise Exception("Cannot detect os version. Exiting...")

lhoss commented on playbooks/roles/database/tasks/main.yml:
```diff
@@ -79,17 +79,32 @@
     name: "{{ item }}"
     update_cache: yes
     state: present
+#    enablerepo: "{{ postgres_enablerepo | default(omit) }}"
```
I forgot to remove that comment .. but since the new feature https://github.com/hortonworks/ansible-hortonworks/commit/de71ccf4918b6e17df4bb297c6b50f8a5b3603cf 
enablerepo should probably also be used for postgres!

lhoss replied:
comment removed and added commit
> external postgres repository only used if (new) option add_repo enabled

alexandruanghel commented:
Hi, I only managed to test this and it looks good, thanks for the contribution!

I have a question about one bit which might not be needed: https://github.com/hortonworks/ansible-hortonworks/pull/67/files#diff-17537708d9884da712724464939a41eeR104

Is there a reason you have to execute that as the `bin/initdb` command which requires sudo?
From what I can see in the postgres [docs](https://www.postgresql.org/download/linux/redhat/) you only need to execute `/usr/pgsql-9.6/bin/postgresql96-setup initdb` which doesn't need sudo and looks better/simpler to me (main reason being because it's in the docs).


About the amazon linux, don't worry too much about it :) 
It's not supported in HDP2.6 anyway (although some HDP versions do work on amazon linux 1, but not 2).
HDP 3/Ambari 2.7 was supposed to introduce support for Amazon Linux, however there were some bugs in the initial version and it should only work starting with Ambari 2.7.1.0.

I've already done some work to bring Amazon Linux 2 into these playbooks and will commit soon (I'll also remove support for amazon linux 1 as that was never officially supported by HWX anyway).

lhoss commented:
> Is there a reason you have to execute that as the `bin/initdb` command which requires sudo?
> From what I can see in the postgres [docs](https://www.postgresql.org/download/linux/redhat/) you only need to execute `/usr/pgsql-9.6/bin/postgresql96-setup initdb` which doesn't need sudo and looks better/simpler to me (main reason being because it's in the docs).

Yeah the reason was I couldn't find  postgresql(96)-setup for the redhat-amazon case, so now I
reverted back to use 'postgresql96-setup initdb' (I had tested it initially) and reverted rh-amazon to original/old default postgres (as you will retire it soon anyway, and indeed the amazon-2 linux support is the one I'm also looking fwd too 👍 )

alexandruanghel commented:
resolves #61
